### PR TITLE
[checkbox-ce-oem] add test cases to validate those plug and slot interface defined in gadget snap (New)

### PR DESCRIPTION
**Pull request by stanley31huang**
## Description
add test cases to validate those plug and slot interface defined in gadget snap

## Resolved issues
N/A

## Documentation
N/A

## Tests
Side-loading test results: https://certification.canonical.com/hardware/202406-34081/submission/389585/

### CODE REVIEW

**Review by baconYao — COMMENTED** [review]
How about we put `gadget_interface_schema.json` to `data` folder?

File: contrib/checkbox-ce-oem/checkbox-provider-ce-oem/units/gadget/manifest.pxu [hunk 1]
```diff
@@ -0,0 +1,4 @@
+unit: manifest entry
+id: has_snap_interface_been_defined_in_gadget
+_name: Does platform has plugs and slots been defined in Gadget snap?
```
Comment (baconYao) [inline contrib/checkbox-ce-oem/checkbox-provider-ce-oem/units/gadget/manifest.pxu:3]:
```suggestion
_name: Does platform have plugs and slots been defined in Gadget snap?
```

File: contrib/checkbox-ce-oem/checkbox-provider-ce-oem/units/gadget/jobs.pxu [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    Usage GADGET_INTERFACE_FILE=/home/user/gadget_interface.json
+command:
+    echo "Check gadget interface file been defined in GADGET_INTERFACE_FILE"
+    [ -f "$GADGET_INTERFACE_FILE" ] && echo "Available" || echo "Not Available"
+
+id: snap_interface_resource
+_summary: Generates a list of slot that defined in the gadget snap
+_description:
+    Generates a slot list that is defined in GADGET_INTERFACE_FILE variable.
+    Usage
```
Comment (baconYao) [inline contrib/checkbox-ce-oem/checkbox-provider-ce-oem/units/gadget/jobs.pxu:27]: What's the purpose of `Usage` here?

File: contrib/checkbox-ce-oem/checkbox-provider-ce-oem/bin/check_gadget_interface.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    system_snap_interfaces = Snapd().interfaces()[interface_key]
+
+    for interface in copy.deepcopy(system_snap_interfaces):
+        if interface["snap"] != gadget_snap:
+            system_snap_interfaces.remove(interface)
+
+    return system_snap_interfaces
+
+
+def varify_slot_interface(name, interface):
```
Comment (baconYao) [inline contrib/checkbox-ce-oem/checkbox-provider-ce-oem/bin/check_gadget_interface.py:39]:
```suggestion
def verify_slot_interface(name, interface):
```

File: contrib/checkbox-ce-oem/checkbox-provider-ce-oem/bin/check_gadget_interface.py [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+            exit(0)
+        else:
+            logging.error("The slot interface type of %s mismatch", name)
+            exit(1)
+
+    logging.error("The %s slot interface is not defined in gadget", name)
+    exit(1)
+
+
+def varify_plug_interface(name, interface):
```
Comment (baconYao) [inline contrib/checkbox-ce-oem/checkbox-provider-ce-oem/bin/check_gadget_interface.py:64]:
```suggestion
def verify_plug_interface(name, interface):
```

File: contrib/checkbox-ce-oem/checkbox-provider-ce-oem/bin/check_gadget_interface.py [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+
+
+def main():
+    args = register_arguments()
+
+    if args.type == "slot":
+        # AI: to verify the slot attrs
+        varify_slot_interface(args.name, args.interface)
+    elif args.type == "plug":
+        varify_plug_interface(args.name, args.interface)
```
Comment (baconYao) [inline contrib/checkbox-ce-oem/checkbox-provider-ce-oem/bin/check_gadget_interface.py:98]:
```suggestion
        verify_plug_interface(args.name, args.interface)
```

File: contrib/checkbox-ce-oem/checkbox-provider-ce-oem/bin/check_gadget_interface.py [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+    logging.error("The %s plug interface is not defined in gadget", name)
+    exit(1)
+
+
+def main():
+    args = register_arguments()
+
+    if args.type == "slot":
+        # AI: to verify the slot attrs
+        varify_slot_interface(args.name, args.interface)
```
Comment (baconYao) [inline contrib/checkbox-ce-oem/checkbox-provider-ce-oem/bin/check_gadget_interface.py:96]:
```suggestion
        verify_slot_interface(args.name, args.interface)
```

**Review by baconYao — CHANGES_REQUESTED** [review]

File: contrib/checkbox-ce-oem/checkbox-provider-ce-oem/units/gadget/jobs.pxu (hunk 2 above)
Comment (stanley31huang) [inline contrib/checkbox-ce-oem/checkbox-provider-ce-oem/units/gadget/jobs.pxu:27]:
```suggestion
```

Comment (stanley31huang) [reply contrib/checkbox-ce-oem/checkbox-provider-ce-oem/units/gadget/jobs.pxu:27]: remove unused description

Comment (stanley31huang) [reply contrib/checkbox-ce-oem/checkbox-provider-ce-oem/units/gadget/jobs.pxu]: removed it

Comment (stanley31huang) [inline contrib/checkbox-ce-oem/checkbox-provider-ce-oem/units/gadget/jobs.pxu]:
```suggestion
```

**Review by baconYao — APPROVED** [review]
LGTM

**Pull request merged**